src/components/button/index.tsx

```diff
@@ -0,0 +1,64 @@
+import { ButtonProps } from './types'
+import s from './button.module.css'
+import classNames from 'classnames'
+
+const Button = ({
+  ariaLabel,
+  ariaLabelledBy,
+  ariaDescribedBy,
+  color = 'primary',
+  disabled,
+  icon,
+  iconPosition = 'leading',
+  isFullWidth,
+  onClick,
+  size = 'medium',
+  text,
+  type = 'button',
+}: ButtonProps) => {
+  const className = classNames(s.root, s[size], s[color], {
+    [s.fullWidth]: isFullWidth,
+  })
+  const hasIcon = !!icon
+  const hasLeadingIcon = hasIcon && iconPosition === 'leading'
+  const hasTrailingIcon = hasIcon && iconPosition === 'trailing'
+  const hasText = !!text
+  const hasLabel = !!ariaLabel || !!ariaLabelledBy || !!ariaDescribedBy
+
+  if (!hasIcon && !hasText) {
+    throw new Error(
+      '`Button` must have either `text` or an `icon` with accessible labels.'
+    )
+  }
+
+  if (hasIcon && !hasLabel) {
```

Comment:
Noticed the Swingset examples for `Icon and Text` are throwing this error. If we have text in the button, then I think we don't need the label, right?

```suggestion
  if (hasIcon && !hasLabel && !hasText) {
```

Reply:
Uh oh! I didn't notice the error message pop up below these, will work on fixing it 😊

Reply:
Fixed in https://github.com/hashicorp/dev-portal/pull/215/commits/c9818e981f0c32e705dceaeb7eb6a142702fd844! Added an `isIconOnly` boolean and use it in this check now. Great catch! ✨